src/lib/logger.ts

```diff
@@ -0,0 +1,364 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { Writable } from 'stream';
+import util from 'util';
+import winston from 'winston';
+import Transport from 'winston-transport';
+
+interface VIPLoggerOptions {
+	logDir?: string;
+	consoleLevel?: string;
+	fileLevel?: string;
+}
+
+interface DebugLikeLogger {
+	( ...args: unknown[] ): boolean;
+	enabled: boolean;
+	namespace: string;
+	error: ( ...args: unknown[] ) => boolean;
+	warn: ( ...args: unknown[] ) => boolean;
+	info: ( ...args: unknown[] ) => boolean;
+	verbose: ( ...args: unknown[] ) => boolean;
+	debug: ( ...args: unknown[] ) => boolean;
+	silly: ( ...args: unknown[] ) => boolean;
+}
+
+class VIPLogger {
+	private logDir: string;
+	private rootLogger: winston.Logger;
+	private loggers: Map< string, DebugLikeLogger >;
+	private enabledNamespaces: Set< string >;
+	private isDebugMode: boolean;
+	private commandName: string;
+	private logFilePath: string;
+
+	constructor( options: VIPLoggerOptions = {} ) {
+		this.logDir = options.logDir || path.join( os.homedir(), '.vip-cli', 'logs' );
+		fs.mkdirSync( this.logDir, { recursive: true } );
+
+		// Check if --debug flag is passed
+		this.isDebugMode = process.argv.includes( '--debug' ) || process.argv.includes( '-d' );
+
+		// Container for namespace-specific loggers
+		this.loggers = new Map();
+
+		// Parse DEBUG env var like the debug library does
+
+		this.enabledNamespaces = new Set( [ '*' ] ); // Enable all by default
+		if ( process.env.DEBUG ) {
+			this.enabledNamespaces = this._parseDebugEnv( process.env.DEBUG );
+		}
+
+		// Determine command name from process.argv
+		this.commandName = this._determineCommandName();
+
+		// Create log file path with timestamp and command name
+		this.logFilePath = this._createLogFilePath();
+
+		// Create the root logger with shared transports
+		this.rootLogger = this._createRootLogger( options );
+	}
+
+	private _determineCommandName(): string {
+		// Get the command name from the process arguments
+		const args = process.argv.slice( 2 );
+
+		// If no args, use 'vip' as the default command name
+		if ( args.length === 0 ) {
+			return 'vip';
+		}
+
+		// Extract the command name (first non-flag argument)
+		let cmdName = 'vip';
+		for ( const arg of args ) {
+			if ( ! arg.startsWith( '-' ) ) {
+				cmdName = arg;
+				break;
+			}
+		}
+
+		// If the process is running a specific bin file like vip-dev-env-start.js,
+		// extract the command name from the filename
+		const processPath = process.argv[ 1 ] || '';
+		const match = processPath.match( /vip-([\w-]+)\.js$/ );
```

Comment:
```suggestion
		const match = /vip-([\w-]+)\.js$/.exec( processPath );
```

> `String.match()` behaves the same way as `RegExp.exec()` when the regular expression does not include the global flag `g`. While they work the same, `RegExp.exec()` can be slightly faster than `String.match()`. Therefore, it should be preferred for better performance.

Reply:
I'm gonna rework that function anyway, it's weird.